File: java/minimum_depth_of_binary_tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int minDepth(TreeNode root) {
        if (root==null) return 0;
        if (root.left==null && root.right==null) return 1;
        int mindepth=Integer.MAX_VALUE;
        if (root.left!=null) {
            mindepth=Math.min(mindepth, minDepth(root.left)+1);
        }
        if (root.right!=null) {
            mindepth=Math.min(mindepth, minDepth(root.right)+1);
        }
        return mindepth;        
    }
}

File: cpp/minimum-depth-of-binary-tree.cpp
111. Minimum Depth of Binary Tree  
Given a binary tree, find its minimum depth.

The minimum depth is the number of nodes along the shortest path from the root node down to the nearest leaf node.

Hide Tags Tree Depth-first Search Breadth-first Search
Hide Similar Problems (E) Binary Tree Level Order Traversal (E) Maximum Depth of Binary Tree

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        if (root==NULL) return 0;
        int depth=INT_MAX;
        if (root->left) depth=min(depth, minDepth(root->left));
        if (root->right) depth=min(depth, minDepth(root->right));
        return depth==INT_MAX?1:depth+1;
    }
};
